src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -93,15 +94,26 @@ public override bool Execute()
             return true;
         }
 
-        public string GetCurrentRevisionForBuildName(string buildId)
+        public bool SetVersionAndRevisionFromBuildId(string buildId)
         {
-            Regex regex = new Regex(@"\.(\d+)$");
+            Regex regex = new Regex(@"(\d{8})\.(\d+)$");
+            string dateFormat = "yyyyMMdd";
             Match match = regex.Match(buildId);
-            if (match.Success && match.Groups.Count > 1)
+            if (match.Success && match.Groups.Count > 2)
             {
-                return match.Groups[1].Value;
+                DateTime buildIdDate;
+                if (!DateTime.TryParseExact(match.Groups[1].Value, dateFormat, enUS, DateTimeStyles.AssumeLocal, out buildIdDate))
+                {
+                    Log.LogError("The OfficialBuildId doesn't follow the expected({0}.rr) format: '{1}'", dateFormat, match.Groups[1].Value);
+                    return false;
+                }
+                buildIdDate = buildIdDate.ToUniversalTime();
+                GeneratedVersion = GetCurrentVersionForDate(buildIdDate, ComparisonDate);
+                GeneratedRevision = match.Groups[2].Value;
```

Comment:
We do for corefx, but this code is intended to be used by different repos, so we decided to not  have that constrain on the BuildNumberMinor as some other consumer might want the rev to be only 1 digit, or more than 2 if they produce more than 99 builds a day.